csaf-retrieval/src/jvmMain/kotlin/io/github/csaf/sbom/retrieval/RetrievedDocument.kt

```diff
@@ -61,5 +62,13 @@ data class RetrievedDocument(val json: Csaf, @Suppress("unused") val sourceUrl:
                     throw Exception("Failed to load CSAF document from $documentUrl", e)
                 }
         }
+
+        fun fromJson(json: String): ResultCompat<RetrievedDocument> {
+            return try {
+                ResultCompat.success(RetrievedDocument(Json.decodeFromString<Csaf>(json)))
```

Comment:
Any particular reason not to expose `Json.decodeFromStream`?

Reply:
Stream is JVM-specific, but I included both now.

Reply:
It's experimental, BTW. Living on the edge. :laughing: